src/gadgets/sumcheck.rs

```diff
@@ -0,0 +1,336 @@
+#![allow(unused)]
```

Comment:
Can we add some basic documentation to this gadget?

Reply:
yes, agree. @CPerezz also mentioned it (https://github.com/privacy-scaling-explorations/Nova/pull/28#discussion_r1263520355) and is planned once the impl is more solid.